metadata-db-app/src/cmr/metadata_db/data/oracle/granule_table.clj

```diff
@@ -30,20 +24,16 @@
         transaction_id INTEGER DEFAULT 0 NOT NULL,
         created_at TIMESTAMP WITH TIME ZONE DEFAULT SYSTIMESTAMP NOT NULL"))
 
-(defmethod granule-column-sql true
+(defn granule-column-sql
+  "Returns the sql to define provider granule columns"
   [provider]
-  ;; For small provider granule table, there is an extra provider_id column
-  (str (granule-column-sql {:small false})
-       ",provider_id VARCHAR(255) NOT NULL"))
-
-(defmulti granule-constraint-sql
-  "Returns the sql to define constraint on provider granule table"
-  (fn [provider table-name]
-    (:small provider)))
+  (case (:small provider)
+    false granule-column-sql-false
+    true (str granule-column-sql-false ",provider_id VARCHAR(255) NOT NULL")))
 
-(defmethod granule-constraint-sql false
-  [provider table-name]
-  (util/validate-table-name table-name)
+(defn- granule-constraint-sql-false
```

Comment:
Is there a name without a boolean value in it that would work well?

Reply:
Suggestions:
- create-granule-constraint-sql-non-small-prov
- get-granule-constraint-sql-non-small-prov

Reply:
Yes, I changed the functions to use create-...   and nothing for the normal providers and I have added 
create-...-small-provider for the functions that represent the small provider.